modules/android/pages/upgrade.adoc

```diff
@@ -14,341 +14,381 @@
 // CBL-C started at version 3.0
 [IMPORTANT]
 --
-On upgrading from a 2.x release, all Couchbase Lite databases will be automatically re-indexed on initial database open. +
+On upgrading from a 2.x release, all Couchbase Lite databases automatically re-indexes on initial database open. +
 This can result in a delay before the database is usable.
 --
 
-[#3-2-3-upgrade]
+[#4-0-0-upgrade]
 == {major}.{minor}.{base}{empty} Upgrade
 
-The action will take place automatically and can lead to some delay in the database becoming available for use in your application.
+Couchbase Lite 4.0 introduces significant architectural changes, most notably the migration from revision trees to version vectors for document versioning.
 
-In addition, if you are syncing with a {major}.{minor}.{base}{empty} Sync Gateway, you should be aware of the significant configuration enhancements introduced and their impact.
-See xref:sync-gateway::upgrading.adoc[Upgrading Sync Gateway] for more details.
-This is a one-way conversion.
 
-// All API and other listed changes included in the other platforms do not apply to CBL-C for the following reasons
-// The changes apply to pre 3.x versions (CBL-C started at 3.0)
-// Or they relate to Querybuilder functionality - CBL-C doesn't have Querybuilder.
-[#api-changes]
-=== API Changes
+=== Major Changes in {major}.{minor}.{base}{empty}
 
+**Version Vector Architecture**:
+CBL {major}.{minor}.{base}{empty} replaces the revision tree system with version vectors, providing improved performance, scalability, and conflict resolution. Documents now use version-based revision IDs in the format `<timestamp>@<source-id>` instead of the previous `<generation>-<document-hash>` format.
 
-This content introduces the changes made to the Couchbase{nbsp}Lite for Android API for release {major}.{minor}.{base}{empty}.
+**Enhanced Conflict Resolution**:
+the default conflict resolution strategy changes from `most active wins` to `last write wins` based on hybrid logical timestamps, providing more intuitive and predictable conflict resolution behavior.
```

Comment:
`The` instead of `the` default conflict ... ?

Reply:
Not really, we use small letters in a list after a list header so The will not comply with vale and the style guide.

See screenshot below 

<img width="740" height="73" alt="Screenshot 2025-09-19 at 5 11 47 pm" src="https://github.com/user-attachments/assets/d206f243-22de-48a5-a602-466db5a67a13" />